plotly/tests/test_optional/test_figure_factory.py

```diff
@@ -1267,6 +1267,15 @@ def test_scatter_plot_matrix_kwargs(self):
                                exp_scatter_plot_matrix['layout'])
 
 
+class TestGantt(NumpyTestUtilsMixin, TestCase):
+
+    def test_df_dataframe(self):
+
+        # validate dataframe has correct column names
+        df1 = pd.DataFrame([[2, 'Apple']], columns=['Numbers', 'Fruit'])
+        self.assertRaises(PlotlyError, tls.FigureFactory.create_gantt, df1)
```

Comment:
> Doesn't look like you have a test that indeed inputs a dataframe. The other test in test_core seems to insert a list of dict's. Can you write a 🐅 that successfully uses a dataframe to create a gannt chart?

Sure. How thorough should these sets of tests be? Or are you thinking of an `all_args` kind of thing that works?